src/drivers/cholqrcp.cc

```diff
@@ -0,0 +1,249 @@
+#include <cstdint>
+#include <vector>
+
+#include <RandBLAS.hh>
+#include <lapack.hh>
+#include <RandLAPACK.hh>
+
+#include <chrono>
+
+using namespace RandLAPACK::comps::util;
+using namespace blas;
+using namespace lapack;
+
+using namespace std::chrono;
+
+namespace RandLAPACK::drivers::cholqrcp {
+
+// This vesrion of the code overwrites matrix A with Q
+// Note that Q is now useless here
+template <typename T>
+int CholQRCP<T>::CholQRCP1(
+    int64_t m,
+    int64_t n,
+    std::vector<T>& A,
+    int64_t d,
+    std::vector<T>& R,
+    std::vector<int64_t>& J
+){
+    //-------TIMING VARS--------/
+    high_resolution_clock::time_point saso_t_stop;
+    high_resolution_clock::time_point saso_t_start;
+    long saso_t_dur = 0;
+
+    high_resolution_clock::time_point qrcp_t_start;
+    high_resolution_clock::time_point qrcp_t_stop;
+    long qrcp_t_dur = 0;
+    
+    high_resolution_clock::time_point rank_reveal_t_start;
+    high_resolution_clock::time_point rank_reveal_t_stop;
+    long rank_reveal_t_dur = 0;
+
+    high_resolution_clock::time_point cholqrcp_t_start;
+    high_resolution_clock::time_point cholqrcp_t_stop;
+    long cholqrcp_t_dur = 0;
+
+    high_resolution_clock::time_point a_mod_piv_t_start;
+    high_resolution_clock::time_point a_mod_piv_t_stop;
+    long a_mod_piv_t_dur = 0;
+
+    high_resolution_clock::time_point a_mod_trsm_t_start;
+    high_resolution_clock::time_point a_mod_trsm_t_stop;
+    long a_mod_trsm_t_dur = 0;
+
+    high_resolution_clock::time_point copy_t_start;
+    high_resolution_clock::time_point copy_t_stop;
+    long copy_t_dur = 0;
+
+    high_resolution_clock::time_point resize_t_start;
+    high_resolution_clock::time_point resize_t_stop;
+    long resize_t_dur = 0;
+
+    high_resolution_clock::time_point total_t_start;
+    high_resolution_clock::time_point total_t_stop;
+    long total_t_dur = 0;
+    //-------TIMING VARS--------/
+
+    //-------TIMING--------/
+    if(this -> timing) {
+        total_t_start = high_resolution_clock::now();
+    
+        resize_t_start = high_resolution_clock::now();
+    }
+    //-------TIMING--------/
+
+    T* A_dat       = A.data();
+    T* A_hat_dat   = upsize(d * n, this->A_hat);
+    T* tau_dat     = upsize(n, this->tau);
+    J.resize(n);
+    int64_t* J_dat = J.data();
+
+    //-------TIMING--------/
+    if(this -> timing) {
+        resize_t_stop = high_resolution_clock::now();
+        resize_t_dur = duration_cast<microseconds>(resize_t_stop - resize_t_start).count();
```

Comment:
Style option: if you're going to define all these variables at the start of the function, how about you wait to compute any of the ``_dur`` ("duration") variables until the end of the function? That way more of these ``if( this->timing )`` statements can be reduced to one line, and we can remove the brackets.

Reply:
Alternatively, if you decide to make a struct that encapsulates all of these variables, then you can define an instance method of the struct so that all of these reduction operations happen automatically with a single function call.

Reply:
Resolved this using the "Style option" comment. Using a struct, in my opinion, would not simplify much.